internal/date/date.go

```diff
@@ -0,0 +1,36 @@
+// Package date provides functions for working with dates and times in the context of DIMO Rewards.
+package date
```

Comment:
probably a task for another day but moving this into shared could be something to add to the backlog

Reply:
I think these days we’re more likely to have a freestanding repo

Reply:
I also though of moving this out of internal so others could import from this repo if they'd like